api/app/controllers/spree/api/orders_controller.rb

```diff
@@ -14,6 +14,12 @@ class OrdersController < Spree::Api::BaseController
         end
       end
 
+      def approve
+        authorize! :update, @order, params[:token]
```

Comment:
:bomb: This isn't a sufficient check. All users have the ability to update their own order. Only admins should be able to approve an order.